### ISSUE TITLE
Some questions for plugins in python

### ISSUE THREAD

**Issue by Pagalom** [Mahlet-Inc/hobbits#84]
Hi,

got some question about plugins in python :

-  In the JSON file, when I put "decimal" or "integer" type for my parameter, it doesn't change anything, in the both cases it's an integer, is it normal ? Do i miss something ?

-  Can we put negative values in our integer field ? And so, can we precise a maximum and a minimum value in the JSON file for a specific parameter ?

-  When I try to add a boolean parameter in my JSON file and execute my plugin in Hobbits, I've always this error :

```bash
"Operator Plugin Error: Plugin 'test' reported an error with its processing: Other errors:
Failed to parse arg 4"
(here the arg 4 is of course my boolean, if i put more parameters, it will say 'arg 5' or 'arg 6')
```
and this is my script in python :
```python
def operate_on_bits(input, output_bits, output_info, perso_var, progress):
  if input.bits.size() > 300 and perso_var:  # the perso_var is my boolean
    output_info.add_highligt("python", "mychunck", 20, 3000)

  output_bits.resize(input.bits.size())
  for i in range(0, input.bits.size()):
    output_bits.set(i, input.bits.at(i))  # this is the only other modification in the default plugin (I removed the 'not')
    if progress.is_cancelled():
      return
    progress.set_progress(i, input.bit.size())
```
and my JSON file :
```json
{
  "name": "test",
  "description": "plugin test",
  "tags": ["test"],
  "script": "main.py",
  "type": "operator",
  "extra_paths": [],
  "parameters": [
    {
      "name": "perso_var",
      "type": "boolean"
    }
  ]
}
```

**Comment by hello-adam** [comment]
I'm so sorry, I missed this!

> * In the JSON file, when I put "decimal" or "integer" type for my parameter, it doesn't change anything, in the both cases it's an integer, is it normal ? Do i miss something ?

It looks like it will make an integer spinbox in both cases, which is not the intended behavior. I think I'll be able to address this soon.

> * Can we put negative values in our integer field ? And so, can we precise a maximum and a minimum value in the JSON file for a specific parameter ?

Yes, right now it looks like it defaults to `0-0x7fffffff` (e.g. `spinBox->setRange(0, 0x7fffffff);`), but if you add "min" and/or "max" parameters you can get around that. FOr example, here's a power spectral display plugin config I've been messing around with
```
{
    "name": "PyPSD",
    "description": "Python Power Spectral Density display using matplotlib",
    "tags": ["Python"],
    "script": "plugin.py",
    "type": "display",
    "extra_paths": [],
    "render_config": {
        "asynchronous": true,
        "hide_bit_offset_controls": true
    },
    "parameters": [
        {
            "name": "FFT Size",
            "type": "integer",
            "min": 16,
            "max": 32768
        },
        {
            "name": "FFT Number",
            "type": "integer",
            "min": 1
        },
        {
            "name": "Sample Frequency (Hz)",
            "type": "integer",
            "min": 1
        }
    ]
}

```

> * When I try to add a boolean parameter in my JSON file and execute my plugin in Hobbits, I've always this error :
> 
> ```shell
> "Operator Plugin Error: Plugin 'test' reported an error with its processing: Other errors:
> Failed to parse arg 4"
> (here the arg 4 is of course my boolean, if i put more parameters, it will say 'arg 5' or 'arg 6')
> ```

Yes this is another bug in the configuration interface.

Thank you for reporting these! I think I wrote the documentation aspirationally and then never finished implementing stuff. I can try to get quick fixes for these soon.

**Comment by hello-adam** [comment]
I think cef1055 fixes the bugs mentioned here, but I'm going to wait until I update the documentation and maybe do a little more improvement/validation before pushing a release. The release should come this weekend or next week.

**Comment by Pagalom** [comment]
I just saw your answers, tanks a lot ! It will help.

**Comment by hello-adam** [comment]
Alright, the fixes and additions should all be in 0.47.0

Let me know if anything else goes wrong or if you have any ideas about the python plugin interface.

**Comment by Pagalom** [comment]
Thanks à lot ! I'll try it soon and keep you up to date
